dwa_local_planner/src/dwa_planner.cpp

```diff
@@ -285,9 +285,38 @@ namespace dwa_local_planner {
     // robot needs to make a 180 degree turn at the end
     std::vector<geometry_msgs::PoseStamped> front_global_plan = global_plan_;
     double angle_to_goal = atan2(goal_pose.pose.position.y - pos[1], goal_pose.pose.position.x - pos[0]);
+
+    // reduce crop forward_point_distance such that the robot can reach the goal pose
+    // without its nose entering space considered as occupied by obstacles
+    double forward_point_distance = 0;
+    const auto cos_angle_to_goal = cos(angle_to_goal);
+    const auto sin_angle_to_goal = sin(angle_to_goal);
+    const int res = 10;
+    for (int i = 1; i <= res; ++i) {
```

Comment:
ideally we don't want to waste loops in checking from closest to farthest until we meet obstacle as this is usually a off case. If the forward point distance is blocked then we can check from nearest to furthest but otherwise I think we should skip this loop. yes this is not technically great as there might be an obstacle infront but it might be free at forward_point_distance but I think its worth saving on the computation for the rare case.

Reply:
I was basically going through the thought process in your comment, but my judgment here was that I'd rather prevent the "not technically great" edge cases (specifically, the cost in front of the robot is evaluated for all points along a trajectory, so any invalid point between goal and the point `forward_point_distance` in front of the goal, albeit rare, would invalidate the trajectory) and have simpler code. Also this loop is executed once for a planning cycle, so it's computation will be negligible compared to all the other stuff that happens in an overall planning cycle.